pwndbg/dbg/__init__.py

```diff
@@ -12,6 +13,169 @@
+        """
+        What category of type this object belongs to.
+        """
+        raise NotImplementedError()
+
+    def fields(self) -> List[TypeField] | None:
+        """
+        List of all fields in this type, if it is a structured type.
+        """
+        raise NotImplementedError()
+
+    def array(self, count: int) -> Type:
+        """
+        Return a type that corresponds to an array whole elements have this type.
+        """
+        raise NotImplementedError()
+
+    def pointer(self) -> Type:
+        """
+        Return a pointer type that has this type as its pointee.
+        """
+        raise NotImplementedError()
+
+    def strip_typedefs(self) -> Type:
+        """
+        Return a type that corresponds to the base type after a typedef chain,
+        if this is a typedef. Returns the type itself otherwise.
+        """
+        raise NotImplementedError()
+
+    def target(self) -> Type:
+        """
+        Return the target of this reference type, if this is a reference type.
+        """
+        raise NotImplementedError()
+
+
+class Value:
+    """
+    Class representing a value in the context of an inferior process.
+    """
+
+    @property
+    def address(self) -> Value | None:
+        """
+        The address of this value, in memory, if addressable, otherwise `None`.
+        """
+        raise NotImplementedError()
+
+    @property
+    def is_optimized_out(self) -> bool:
+        """
+        Whether this value is present in debugging information, but has been
+        optimized out of the actual program.
+        """
+        raise NotImplementedError()
+
+    @property
+    def type(self) -> Type:
+        """
+        The type associated with this value.
+        """
+        raise NotImplementedError()
+
+    def dereference(self) -> Value:
+        """
+        If this is a poitner value, dereferences the pointer and returns a new
+        instance of Value, containing the value pointed to by this pointer.
+        """
+        raise NotImplementedError()
+
+    # The intent of this function has a great deal of overlap with that of
+    # `pwndbg.gdblib.memory.string()`. It probably makes sense to take this
+    # functionality out of the debugger API.
+    #
+    # TODO: Move to single, common string function.
+    def string(self) -> str:
+        """
+        If this value is a string, then this method converts it to a Python string.
+        """
+        raise NotImplementedError()
+
+    # This is a GDB imeplementation detail.
+    def fetch_lazy(self) -> None:
+        """
+        Fetches the value if it is lazy, does nothing otherwise.
+        """
+        raise NotImplementedError()
+
+    def __int__(self) -> int:
+        """
+        Converts this value to an integer, if possible.
+        """
+        raise NotImplementedError()
+
+    # Because casting is still sloppy (i.e. it acceps `gdb.Type` objects) in
+    # some places, we have to allow it here for lints to pass.
```

Comment:
```suggestion
    # Because casting is still sloppy (i.e. it accepts `gdb.Type` objects) in
    # some places, we have to allow `Any` here for lints to pass.
```